oracle/types/market.go

```diff
@@ -1,160 +1,66 @@
-	}
-
-	if len(tickerConfigs) == 0 {
-		return ProviderMarketMap{
-			Name:          name,
-			TickerConfigs: make(map[mmtypes.Ticker]mmtypes.ProviderConfig),
-			OffChainMap:   make(map[string]mmtypes.Ticker),
-		}, nil
-	}
-
-	offChainMap := make(map[string]mmtypes.Ticker)
-	for ticker, config := range tickerConfigs {
-		if err := ticker.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid ticker %s: %w", ticker, err)
-		}
-
-		if err := config.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid provider config for %s: %w", ticker, err)
-		}
-
-		if config.Name != name {
-			return ProviderMarketMap{}, fmt.Errorf("expected provider config name %s, got %s", name, config.Name)
-		}
-
-		offChainMap[config.OffChainTicker] = ticker
-	}
-
-	return ProviderMarketMap{
-		Name:          name,
-		TickerConfigs: tickerConfigs,
-		OffChainMap:   offChainMap,
-	}, nil
-}
-
-// GetTickers returns the tickers from the provider market map.
-func (pmm *ProviderMarketMap) GetTickers() []mmtypes.Ticker {
-	tickers := make([]mmtypes.Ticker, 0, len(pmm.TickerConfigs))
-	for ticker := range pmm.TickerConfigs {
-		tickers = append(tickers, ticker)
-	}
-	return tickers
-}
-
-// ValidateBasic performs basic validation on the provider market map.
-func (pmm *ProviderMarketMap) ValidateBasic() error {
-	if len(pmm.Name) == 0 {
-		return fmt.Errorf("provider name cannot be empty")
-	}
-	if len(pmm.OffChainMap) != len(pmm.TickerConfigs) {
-		return fmt.Errorf("off-chain map length mismatch")
-	}
-
-	for ticker, config := range pmm.TickerConfigs {
-		if err := ticker.ValidateBasic(); err != nil {
-			return fmt.Errorf("invalid ticker %s: %w", ticker, err)
-		}
-
-		if err := config.ValidateBasic(); err != nil {
-			return fmt.Errorf("invalid provider config for %s: %w", ticker, err)
-		}
-
-		t, ok := pmm.OffChainMap[config.OffChainTicker]
-		if !ok {
-			return fmt.Errorf("off-chain ticker %s not found in off-chain map", config.OffChainTicker)
-		}
-
-		if t != ticker {
-			return fmt.Errorf("off-chain ticker %s does not match on-chain ticker %s", config.OffChainTicker, ticker)
+			providerTicker := NewProviderTicker(
+				cfg.Name,
+				cfg.OffChainTicker,
+				cfg.Metadata_JSON,
+				DefaultTickerDecimals,
+			)
+			providerTickers = append(providerTickers, providerTicker)
+			seenOffChainTickers[cfg.OffChainTicker] = struct{}{}
 		}
 	}
 
-	return nil
+	return providerTickers, nil
 }
 
-// ReadMarketConfigFromFile reads a market map configuration from a file at the given path.
-func ReadMarketConfigFromFile(path string) (mmtypes.MarketMap, error) {
-	// Initialize the struct to hold the configuration
-	var config mmtypes.MarketMap
+// TickersToProviderTickers is a map of tickers to provider tickers. This should be
+// utilized by providers to configure the tickers they will be providing data for.
+type TickersToProviderTickers map[mmtypes.Ticker]DefaultProviderTicker
```

Comment:
I think it probably makes more sense for this to be.  We can just use the interface

```suggestion
type TickersToProviderTickers map[mmtypes.Ticker]ProviderTicker
```

Reply:
I need this to be the concrete type since I am going to be explicitly assigning DefaultProviderTicker. I would check out the next PR.

Reply:
Why do we need this at all? Shouldn't each provider just be able to take the `[]ProviderTicker`? My intuition is that, ideally, there's no state per ticker in each provider. All info per ticker a provider needs should be accessible from the `ProviderTicker` iface. Reason being, we'll be changing metadata for each ticker (suppose we change decimals or `Metadata_JSON` for a ticker). it's much simpler for this to just work (pass `ProviderTickers`), w/o awareness of each provider internal (having to selectively update providers' ticker state).

Reply:
each provider does take in `[]ProviderTicker`. check out the follow up PRs.

Reply:
the handlers are all essentially stateless now, except for the use of `types.ProviderTickers` which providers some basic utility functions.